gcc/ld errors when compiling with EESSI modules

I was trying to compile some C code, and ran into the following issues:
```
$ gcc hello.c -o hello
/usr/bin/ld.gold: --push-state: unknown option
/usr/bin/ld.gold: use the --help option for usage information
collect2: error: ld returned 1 exit status

$ which -a ld.gold
/usr/bin/ld.gold
/bin/ld.gold
```
We somehow need to instruct gcc to use the one from the compatibility layer.

--- a/init/eessi_environment_variables
+++ b/init/eessi_environment_variables
@@ -23,6 +23,11 @@ if [ -d $EESSI_PREFIX ]; then
 
   # Set EPREFIX since that is basically a standard in Gentoo Prefix
   export EPREFIX=$EESSI_PREFIX/compat/$EESSI_OS_TYPE/$EESSI_CPU_FAMILY
+
+  # add location of commands provided by compat layer to $PATH;
+  # see https://github.com/EESSI/software-layer/issues/52
+  export PATH=$EPREFIX/usr/bin:$PATH
+
   export EESSI_EPREFIX=$EPREFIX
   if [ -d $EESSI_EPREFIX ]; then
 

